tsdb/engine/tsm1/compact.go

```diff
@@ -640,7 +653,7 @@ func (c *DefaultPlanner) Plan(lastWrite time.Time) ([]CompactionGroup, int64) {
 
 			// Skip compacting this group if there happens to be any lower level files in the
 			// middle.  These will get picked up by the level compactors.
-			if lvl <= 3 {
+			if lvl <= 3 && len(gen.files) > 4 {
```

Comment:
I don't understand the length check here.  Why are big groups of lower level files skipped?